src/cozmo/nav_memory_map.py

```diff
@@ -0,0 +1,316 @@
+        #: node. Only leaf nodes have content, this is ``None`` for all other
+        #: nodes.
+        self.content = None  # type: _NodeContentType
+
+        self._next_child = 0  # Used when building to track which branch to follow (
+
+    def __repr__(self):
+        return '<%s center: %s size: %s content: %s>' % (
+            self.__class__.__name__, self.center, self.size, self.content)
+
+    def contains_point(self, x, y):
+        """Test if the node contains the given x,y coordinates.
+
+        Args:
+            x (float): x coordinate for the point
+            y (float): y coordinate for the point
+
+        Returns:
+            bool: True if the node contains the point, False otherwise.
+        """
+        half_size = self.size * 0.5
+        dist_x = abs(self.center.x - x)
+        dist_y = abs(self.center.y - y)
+        return (dist_x <= half_size) and (dist_y <= half_size)
+
+    def get_node(self, x, y, _assumed_in_bounds=False):
+        """Get the node at the given x,y coordinates.
+
+        Args:
+            x (float): x coordinate for the point
+            y (float): y coordinate for the point
+
+        Returns:
+            :class:`NavMemoryMapGridNode`: The smallest node that includes the
+            point. Will be ``None`` if the point is outside of the map.
+        """
+        if not _assumed_in_bounds and not self.contains_point(x, y):
+            # point is out of bounds
+            return None
+
+        if self.children is None:
+            return self
+        else:
+            x_offset = 2 if x < self.center.x else 0
+            y_offset = 1 if y < self.center.y else 0
+            child_node = self.children[x_offset+y_offset]
+            # child node is by definition in bounds / on boundary
+            return child_node.get_node(x, y, True)
+
+    def get_content(self, x, y):
+        """Get the node's content at the given x,y coordinates.
+
+        Args:
+            x (float): x coordinate for the point
+            y (float): y coordinate for the point
+
+        Returns:
+            :class:`_NodeContentType`: The content included at that point.
+            Will be :attr:`NodeContentTypes.Unknown` if the point is outside of
+            the map.
+        """
+        node = self.get_node(x, y)
+        if node:
+            return node.content
+        else:
+            return NodeContentTypes.Unknown
+
+    def _add_child(self, content, depth):
+        """Add a child node to the quad tree.
+
+        The quad-tree is serialized to a flat list of nodes, we deserialize
+        back to a quad-tree structure here, with the depth of each node
+        indicating where it is placed.
+
+        Args:
+            content (:class:`_NodeContentType`):
+            depth (int):
+
+        Returns:
+            bool: True if parent should use the next child for future _add_child
+            calls (this is an internal implementation detail of h
```

Comment:
Good catch - I think I was halfway through writing the docstring for this when I decided I should just rename the function to be private as it's entirely an internal implementation detail, and then because I reviewed the docs based on the generated docs (and private methods automatically don't publish docstrings) I didn't notice that it was still a mess.